ci-operator/step-registry/aws/c2s/init-token-service/aws-c2s-init-token-service-commands.sh

```diff
@@ -0,0 +1,327 @@
+# create secret files
+# ------------------------------
+
+cr_yaml_d=$(mktemp -d)
+echo "extracting CR from image $RELEASE_IMAGE_LATEST"
+
+oc adm release extract ${RELEASE_IMAGE_LATEST} --credentials-requests --cloud=aws --to "${cr_yaml_d}" || exit 1
+
+echo "Extracted CR files:"
+ls $cr_yaml_d
+
+annotation="release.openshift.io/feature-gate: TechPreviewNoUpgrade"
+remove_tech_preview_feature_from_manifests ${cr_yaml_d} ${annotation} || exit 1
+
+credentials_requests_files=`mktemp`
+ls ${cr_yaml_d} > ${credentials_requests_files}
+
+echo "CRs to be processed:"
+cat "${credentials_requests_files}"
+
+while IFS= read -r item
+do
+    name=$(/tmp/yq r "${cr_yaml_d}/${item}" 'spec.secretRef.name')
+    ns=$(/tmp/yq r "${cr_yaml_d}/${item}" 'spec.secretRef.namespace')
+    echo "creating secret file for: \"${ns}\" \"${name}\""
+    create_secret_file ${ns} ${name} ${temp_cred_provider_role} ${key_id} ${key_sec}
+done < ${credentials_requests_files}
+
+# ------------------------------
+# set up token refresh service
+# ------------------------------
+
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-namespace.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cap-token-refresh
+EOF
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-role.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cap-token-refresh
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - patch
+- apiGroups:
+  - cloudcredential.openshift.io
+  resources:
+  - credentialsrequests
+  verbs:
+  - list
+EOF
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-rolebinding.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cap-token-refresh
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cap-token-refresh
+subjects:
+- kind: ServiceAccount
+  name: cap-token-refresh
+  namespace: cap-token-refresh
+EOF
+
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-serviceaccount.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cap-token-refresh
+  namespace: cap-token-refresh
+EOF
+
+
+ca_file=`mktemp`
+cat "${CLUSTER_PROFILE_DIR}/shift-ca-chain.cert.pem" > ${ca_file}
+cat "/var/run/vault/mirror-registry/client_ca.crt" >> ${ca_file}
+cat <<EOF > ${SHARED_DIR}/manifest_cap-token-certs-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cap-certs
+  namespace: cap-token-refresh
+type: Opaque
+stringData:
+  cap.pem: |
+`echo -n ${temp_cred_provider_cert_b64} | base64 -d | sed -e 's/^/    /'`
+  cap.key: |
+`echo -n "${temp_cred_provider_private_key_b64}" | base64 -d | sed -e 's/^/    /'`
+  ca-chain.cert.pem: |
+`cat ${ca_file} | sed -e 's/^/    /'`
+EOF
+
+
+if [ ! -f "${SHARED_DIR}/mirror_registry_url" ]; then
+    echo "File ${SHARED_DIR}/mirror_registry_url does not exist."
+    exit 1
+fi
+
+registry_host=$(head -n 1 ${SHARED_DIR}/mirror_registry_url)
+token_refresh_repo=${registry_host}/staebler/cap-token-refresh
```

Comment:
I don't have a thorough understanding of these systems, but is `staebler` the only repo we have access to? It might be worthwhile to get a dedicated CI repo

Reply:
Updated.

It's trying to [get a new credential](https://github.com/yunjiang29/release/blob/c2s/ci-operator/step-registry/aws/c2s/init-token-service/aws-c2s-init-token-service-commands.sh#L303-L308) from CAP (C2S)/GEOAxIS (SC2S) services every [half an hour](https://github.com/yunjiang29/release/blob/c2s/ci-operator/step-registry/aws/c2s/init-token-service/aws-c2s-init-token-service-commands.sh#L269), and update each related secrets.